fud2/fud-core/src/run.rs

```diff
@@ -92,10 +92,24 @@ impl EmitBuild for EmitBuildFn {
 pub enum ConfigVar {
     /// The key for the config variable.
     Var(String),
+
     /// The key for the config variable followed by the value it should be if the key is not found.
     VarOr(String, String),
 }
 
+/// A shell command tagged with required and generated files.
+#[derive(Debug, Clone)]
+pub struct ShellCmd {
```

Comment:
Maybe this should be called `Rule` or something now… since it's a bit more than a command.